dhis-2/dhis-api/src/main/java/org/hisp/dhis/dataset/Section.java

```diff
@@ -109,7 +112,11 @@ public boolean hasCategoryCombo()
             }
         }
 
-        return categoryCombos;
+        List<CategoryCombo> sortedCategoryombos = categoryCombos.stream().collect( Collectors.toList() );
```

Comment:
I think the sort ordering here is data entry form specific, and so we should sort in the data entry app, and not directly in the Section object class. Makes sense to remove the sorting from here and into data entry?

Reply:
not sure if this is data entry specific ... we are just sorting by size and name not really interested where this can be used.

Reply:
Alright fair enough.